plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/functions/tablechanges/TableChangesFunctionProcessor.java

```diff
@@ -148,7 +148,11 @@ private TableFunctionProcessorState processCdfFile()
                     currentVersionCommitTimestampAsBlock, page.getPositionCount());
             return TableFunctionProcessorState.Processed.produced(new Page(page.getPositionCount(), resultBlock));
         }
-        return FINISHED;
+        if (deltaLakePageSource.isFinished()) {
+            return FINISHED;
+        } else {
+            return TableFunctionProcessorState.Processed.produced(null);
```

Comment:
The produced method has a requireNonNull check: https://github.com/trinodb/trino/blob/master/core/trino-spi/src/main/java/io/trino/spi/function/table/TableFunctionProcessorState.java#L89

Reply:
yeah sorry for this